### ISSUE TITLE
Confusion matrix no value and many classes

### ISSUE THREAD

**Issue by Mirshal** [ultralytics/ultralytics#6747]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I am here to ask for a solution to the problem of the confusion matrix obtained after testing the dataset itself with a total of 7 classes that were changed in data.yaml.
but the results that came out were 40 classes with a confusion matrix that did not have a value for each class?. making it difficult for me to calculate accuracy, precision, and recall manually. I will help and make me happy if there is a solution. thank you very much for your help.
![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/101452195/f54a99af-97e9-425d-a626-fafc3abb0e9e)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by RizwanMunawar** [comment]
@Mirshal Thanks for sharing the comprehensive information. Could you verify whether your validation images encompass all 40 classes present in your training data? If so, kindly disclose the size of your dataset (number of images).

**Note:** If any class from the training data is absent in the validation data, it would result in the absence of a confusion matrix for that specific class.

Feel free to reach out for any further clarification.

Regards,
Ultralytics Team

**Comment by Mirshal** [comment]
thank you for the answer, here I want to ask again how to bring up the value for each row column of the confusion matrix displayed?, I can't see any of the row columns displaying a definite value. because everything looks empty there are no values, only colors appear. I would really appreciate your answer for my research.

**Comment by Mirshal** [comment]
I have tried to configure the data.yaml in various ways but the values for each class in the confusion matrix cannot appear as well. maybe from you if there is a suggestion or the best solution. it will be very helpful for me in completing this research. thank you.

**Comment by glenn-jocher** [comment]
@Mirshal to address the absence of values in the confusion matrix, ensure that:

1. Your `data.yaml` file accurately reflects the 7 classes.
2. The dataset used for training/testing is correctly labeled and matches the class indices in your `data.yaml`.
3. The pre-trained model is compatible with your dataset, or better, fine-tune it on your dataset.
4. There might be a bug or version issue, so update to the latest YOLOv8 and re-evaluate.

If the problem continues, please open an issue in the YOLOv8 repository with details on your setup for further assistance.

**Comment by hariouat** [comment]
Hello everyone, 
I have the same problem when i trained a model for my custom data. 
[
![confusion_matrix_normalized(2)](https://github.com/ultralytics/ultralytics/assets/117453280/719d6381-9a88-4a9c-ac06-47f24163800e)
i don't know why ? Any help?

**Comment by glenn-jocher** [comment]
@hariouat ensure your `data.yaml` accurately lists your custom classes and the dataset is correctly labeled. If everything is correct and the issue persists, it may be a bug. Please consider updating to the latest YOLOv8 version and validate again. For further assistance, open a new issue on the YOLOv8 repo with full details of your setup and training process. 👍

**Comment by hariouat** [comment]
> @hariouat ensure your `data.yaml` accurately lists your custom classes and the dataset is correctly labeled. If everything is correct and the issue persists, it may be a bug. Please consider updating to the latest YOLOv8 version and validate again. For further assistance, open a new issue on the YOLOv8 repo with full details of your setup and training process. 👍

I checked all my files, i think all it's okay. 
When i trained it with YoloV7, the confusion matrix it's okay but not with YOLOV8.
I will open a new issue. 
Thanks

**Comment by glenn-jocher** [comment]
Glad to hear you've double-checked your files, @hariouat. It seems you might be facing a specific issue with YOLOv8. 🛠️ Please open a new issue with all the relevant details so we can delve deeper into this. Your observations with YOLOv7 vs YOLOv8 provide a good starting point for our investigation. Thank you! 🙏

**Comment by hariouat** [comment]
![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/117453280/352478df-763b-4953-9e16-a97b6bca204a)
Hello @glenn-jocher I thought the issue had been resolved, but it appears that it has not. (both for classification and object detection)

**Comment by glenn-jocher** [comment]
Hello @hariouat, thank you for your persistence. If the confusion matrix still doesn't display the expected values after checking your `data.yaml` and ensuring you have the latest YOLOv8 version, please open a detailed issue on our YOLOv8 repo. Our team will take a closer look to assist you better. 🕵️‍♂️🔍

**Comment by Johan154** [comment]
> ![confusion_matrix](https://private-user-images.githubusercontent.com/117453280/290975274-352478df-763b-4953-9e16-a97b6bca204a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MTI2MjksIm5iZiI6MTcwOTcxMjMyOSwicGF0aCI6Ii8xMTc0NTMyODAvMjkwOTc1Mjc0LTM1MjQ3OGRmLTc2M2ItNDk1My05ZTE2LWE5N2I2YmNhMjA0YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQwODA1MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjI5NTkyZmY1YjY4ODRhMGViNzA3MmQyNjJhMWVmMDdlNzIyYTBjYzMyM2MxZjUxOWNjOTM0YzAzNWQ3ZTk2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tecJNp5UYHRNiPfy69OIej1HoNzfmUrnSPlfpdPWiGw) Hello @glenn-jocher I thought the issue had been resolved, but it appears that it has not. (both for classification and object detection)

@hariouat @Mirshal  I had the same issue. I solved it by downgrading matplotlib from 3.8.0 to 3.7.1

**Comment by glenn-jocher** [comment]
Hey @Johan154 and @Mirshal! 😊 It seems like the confusion matrix issue might be related to the matplotlib version. A user found success by switching from matplotlib 3.8.0 to 3.7.1. You might want to give that a try! Here's how you can do it:

```bash
pip install matplotlib==3.7.1
```

After downgrading, please rerun your evaluation to see if the confusion matrix displays correctly. Let us know how it goes!
